Importing Required Libraries

In [35]:
# Import pandas
import pandas as pd

Reading Dataset

In [36]:
# Read the dataset using pandas
df = pd.read_csv("govtech_readiness_dataset.csv")
print(df.head())

  Country_Code     Country WB_Region Income_Level  Population_millions  \
0          ARG   Argentina       LCR           UM                 45.5   
1          AUS   Australia       EAP            H                 26.5   
2          BGD  Bangladesh       SAR           LM                171.5   
3          BRA      Brazil       LCR           UM                211.1   
4          CAN      Canada       NAM            H                 39.3   

  GTMI_Group_2025 GTMI_Score_2025 CGSI_CoreGovSystems_2025  \
0               A           0.753                    0.789   
1               A           0.985                    0.989   
2               A           0.882                    0.933   
3               A           0.986                     0.99   
4               A           0.949                    0.934   

  PSDI_ServiceDelivery_2025 DCEI_CitizenEngagement_2025 GTEI_Enablers_2025  \
0                     0.907                        0.64              0.677   
1                     0.99

Analyzing The Data

In [37]:
# Overview of the dataset's size and structure
# This returns - (number of rows, number of columns)
df.shape

(45, 16)

In [38]:
# Converts the column names of the DataFrame into a Python list
df.columns.tolist()

['Country_Code',
 'Country',
 'WB_Region',
 'Income_Level',
 'Population_millions',
 'GTMI_Group_2025',
 'GTMI_Score_2025',
 'CGSI_CoreGovSystems_2025',
 'PSDI_ServiceDelivery_2025',
 'DCEI_CitizenEngagement_2025',
 'GTEI_Enablers_2025',
 'EGov_Rank_2024',
 'EGov_Index_2024',
 'Online_Service_Index_2024',
 'Human_Capital_Index_2024',
 'Telecom_Infra_Index_2024']

In [39]:
# Convert score_columns to numeric
score_columns = [
    'Population_millions',
    'GTMI_Score_2025',
    'CGSI_CoreGovSystems_2025',
    'PSDI_ServiceDelivery_2025',
    'DCEI_CitizenEngagement_2025',
    'GTEI_Enablers_2025',
    'EGov_Rank_2024',
    'EGov_Index_2024',
    'Online_Service_Index_2024',
    'Human_Capital_Index_2024',
    'Telecom_Infra_Index_2024'
]

for column in score_columns:
    df[column] = pd.to_numeric(df[column], errors='coerce')

print("Converted all score columns to numeric format")

Converted all score columns to numeric format


In [40]:
# Understand the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45 entries, 0 to 44
Data columns (total 16 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Country_Code                 45 non-null     object 
 1   Country                      45 non-null     object 
 2   WB_Region                    45 non-null     object 
 3   Income_Level                 45 non-null     object 
 4   Population_millions          45 non-null     float64
 5   GTMI_Group_2025              45 non-null     object 
 6   GTMI_Score_2025              44 non-null     float64
 7   CGSI_CoreGovSystems_2025     44 non-null     float64
 8   PSDI_ServiceDelivery_2025    44 non-null     float64
 9   DCEI_CitizenEngagement_2025  44 non-null     float64
 10  GTEI_Enablers_2025           44 non-null     float64
 11  EGov_Rank_2024               45 non-null     int64  
 12  EGov_Index_2024              45 non-null     float64
 13  Online_Service_Index_2

In [41]:
#  Statistical summary of the DataFrame (Transpose)
df.describe().T

,count,mean,std,min,25%,50%,75%,max
Population_millions,45.0,137.088889,292.914092,1.40000,23.00000,52.30000,114.90000,1438.10000
GTMI_Score_2025,44.0,0.818591,0.121507,0.51200,0.76050,0.82700,0.91175,0.99700
CGSI_CoreGovSystems_2025,44.0,0.816886,0.116880,0.53600,0.74800,0.80950,0.90600,1.00000
PSDI_ServiceDelivery_2025,44.0,0.873705,0.113683,0.54100,0.81900,0.89900,0.96825,1.00000
DCEI_CitizenEngagement_2025,44.0,0.739091,0.192093,0.25500,0.63475,0.79350,0.89525,0.99500
GTEI_Enablers_2025,44.0,0.844591,0.120002,0.47100,0.78850,0.86200,0.93600,0.99800
EGov_Rank_2024,45.0,54.622222,44.967306,1.00000,14.00000,47.00000,90.00000,169.00000
EGov_Index_2024,45.0,0.803705,0.158024,0.31109,0.68411,0.84516,0.93262,0.98474
Online_Service_Index_2024,45.0,0.815056,0.151736,0.34195,0.75905,0.84080,0.92219,1.00000
Human_Capital_Index_2024,45.0,0.775619,0.176297,0.32537,0.64579,0.80768,0.91748,1.00000


Based on statistical summary, DCEI has a standard deviation of 0.192, which is the highest among the four GTMI sub-dimensions.

Checking For The Duplicate Values

In [42]:
# Returns the number of duplicate rows in the dataset
df.duplicated().sum()

0

 Checking Missing Values

In [43]:
#Identify any gaps in our data
df.isnull().sum()

Country_Code                   0
Country                        0
WB_Region                      0
Income_Level                   0
Population_millions            0
GTMI_Group_2025                0
GTMI_Score_2025                1
CGSI_CoreGovSystems_2025       1
PSDI_ServiceDelivery_2025      1
DCEI_CitizenEngagement_2025    1
GTEI_Enablers_2025             1
EGov_Rank_2024                 0
EGov_Index_2024                0
Online_Service_Index_2024      0
Human_Capital_Index_2024       0
Telecom_Infra_Index_2024       0
dtype: int64

Handling Missing Values

In [45]:
# Convert '-' into missing values
df = df.replace('-', pd.NA)

# No imputation performed
print("Missing values retained as NaN")

Missing values retained as NaN


Check Countries with Missing GTMI Score

In [44]:
# Countries with missing GTMI scores cannot be included in GTMI-related charts and calculations
missing_gtmi = df[df['GTMI_Score_2025'].isna()]
display(missing_gtmi[['Country']])
print("These countries will be excluded from GTMI charts in Power BI.")

,Country
25,Nicaragua


These countries will be excluded from GTMI charts in Power BI.


Add Helper Columns

In [46]:
# 1. Replaces short region codes with readable names
region_name_map = {
    'ECA': 'Europe & Central Asia',
    'EAP': 'East Asia Pacific',
    'NAM': 'North America',
    'MNA': 'Middle East & N. Africa',
    'LCR': 'Latin Am. & Caribbean',
    'SAR': 'South Asia',
    'AFR': 'Sub-Saharan Africa'
}
df['Region_Full'] = df['WB_Region'].map(region_name_map)

In [47]:
# 2. Give number to order_level for easy sorting
income_order_map = {'H': 1, 'UM': 2, 'LM': 3, 'L': 4}
df['Income_Order'] = df['Income_Level'].map(income_order_map)

In [48]:
# 3. Income Full Name for making it more readable
income_name_map = {'H': 'High income', 'UM': 'Upper-middle', 'LM': 'Lower-middle', 'L': 'Low income'}
df['Income_Full'] = df['Income_Level'].map(income_name_map)

In [49]:
print("Added 3 helper columns: Region_Full, Income_Order, Income_Full")

Added 3 helper columns: Region_Full, Income_Order, Income_Full


Save Clean File

In [50]:
# Save a new CSV file in the same folder 
df.to_csv('govtech_clean_dataset.csv', index=False)

QUICK SUMMARY

In [51]:
# These numbers confirm everything loaded correctly
print(f"  Total countries:         {len(df)}")
print(f"  Countries with GTMI:     {df['GTMI_Score_2025'].notna().sum()}")
print(f"  Countries missing GTMI:  {df['GTMI_Score_2025'].isna().sum()} (Nicaragua)")
print(f"  Group A (leading):       {(df['GTMI_Group_2025']=='A').sum()}")
print(f"  Group B (developing):    {(df['GTMI_Group_2025']=='B').sum()}")
print(f"  High income countries:   {(df['Income_Level']=='H').sum()}")
print(f"  Low income countries:    {(df['Income_Level']=='L').sum()}")

  Total countries:         45
  Countries with GTMI:     44
  Countries missing GTMI:  1 (Nicaragua)
  Group A (leading):       34
  Group B (developing):    10
  High income countries:   19
  Low income countries:    2
